src/mlkem768.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023 Markus Friedl.  All rights reserved.
+ * Copyright (c) 2025 Loganaden Velvindron. All rights reserved.
+ * Copyright (c) 2025 Jaykishan Mutkawoa. All rights reserved.
+ * Copyright (c) 2025 Keshwarsingh "Kavish" Nadan. All rights reserved.
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "includes.h"
+
+#include <sys/types.h>
+
+#include <stdio.h>
+#ifdef HAVE_STDINT_H
+#include <stdint.h>
+#endif
+#include <stdbool.h>
+#include <string.h>
+#include <signal.h>
+#ifdef HAVE_ENDIAN_H
+# include <endian.h>
+#endif
+
+#include "kex.h"
+
+#ifdef DROPBEAR_MLKEM768
+
+#include "libcrux_mlkem768_sha3.h"
+#include "mlkem768.h"
+#include "dbutil.h"
+#include "dbrandom.h"
+
+int
+crypto_kem_mlkem768_keypair(unsigned char *pk, unsigned char *sk)
+{
+	u_char rnd[LIBCRUX_ML_KEM_KEY_PAIR_PRNG_LEN];
```

Comment:
Use `unsigned char` here for constency with other Dropbear code.

Reply:
Fixed.